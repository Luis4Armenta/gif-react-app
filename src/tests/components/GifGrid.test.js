import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Probando el componente <GifGrid />', () => {
  const category = 'cats';
  
  test('Debe de cargar el componente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    
    const wrapper = shallow(<GifGrid category={ category }/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/cualquier-cosa.jpg',
      title: 'Cualquier cosa'
    },
    {
      id: '123',
      url: 'https://localhost/cualquier-cosa.jpg',
      title: 'Cualquier cosa'
    }];
    
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    
    const wrapper = shallow(<GifGrid category={ category }/>);
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  })
  
  
});
