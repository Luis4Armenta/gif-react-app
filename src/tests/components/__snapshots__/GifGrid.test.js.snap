// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifGrid /> Debe de cargar el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    cats
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Probando el componente <GifGrid /> Debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    cats
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier-cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquier-cosa.jpg"
    />
  </div>
</Fragment>
`;
